controller/rpc/scanner.go

```diff
@@ -289,18 +361,18 @@ func ScanImage(scanner string, ctx context.Context, req *share.ScanImageRequest)
 }
 
 func ScanPackage(ctx context.Context, pkgs []*share.ScanAppPackage) (*share.ScanResult, error) {
-	scanner := getAvaliableScanner()
+	scanner := ScannerMgr.acquireScannerAvailability()
 	if scanner == "" {
 		err := fmt.Errorf("No scanner available")
 		log.WithFields(log.Fields{"error": err}).Error()
 		return nil, err
 	}
 
-	client, err := getScannerServiceClient(scanner, true)
+	client, scanner, err := ScannerMgr.getScannerServiceClient(scanner, true)
 	if err != nil {
 		return nil, err
 	}
-	defer decScanningCount(scanner)
+	defer ScannerMgr.releaseScannerAvailability(scanner)
```

Comment:
This line should be moved to after `scanner := ScannerMgr.acquireScannerAvailability()`, otherwise the scanner availability would not be increased again if it returns between two calls.